framework/src/interfaces/SolutionInvalidInterface.C

```diff
@@ -31,6 +31,14 @@ SolutionInvalidInterface::flagInvalidSolutionInternal(InvalidSolutionID invalid_
   return solution_invalidity.flagInvalidSolutionInternal(invalid_solution_id);
 }
 
+void
+SolutionInvalidInterface::flagSolutionWarningInternal(InvalidSolutionID invalid_solution_id) const
+{
+  auto & solution_invalidity = _si_moose_object.getMooseApp().solutionInvalidity();
+  _si_problem.allowInvalidSolution(true);
```

Comment:
I dont think this works? 
if we have both
- an invalid case from flagSolutionInvalid
and a warning from flagSolutionWarning we will make an invalid solution as converged

Reply:
If we have both flagSolutionInvalid and flagSolutionWarning, what's the desired behavior? We should still allow solution converge or not? I would prefer to have the solve running and give invalid solution report in the end

Reply:
if invalid we should not allow the convergence (unless the user set allow_solution_invalid)

Reply:
Addressed